wit/deps/golem/golem-host.wit

```diff
@@ -56,6 +76,46 @@ interface host {
     /// Makes the current worker travel back in time and continue execution from the given position in the persistent
     /// op log.
     set-oplog-index: func(oplog-idx: oplog-index) -> ();
+
+    /// Blocks the execution until the oplog has been written to at least the specified number of replicas, 
+    /// or the maximum number of replicas if the requested number is higher.
+    oplog-commit: func(replicas: u8) -> ();
+
+    /// Marks the beginning and end of an atomic operation. 
+    /// In case of a failure within the region selected by the lifecycle of `atomic-operation`,
+    /// the whole region will be reexecuted on retry. 
+    /// The end of the region is either when the resource is dropped, or when the explicit `commit` method is called on it.
+    resource atomic-operation {
+        constructor();
+
+        /// Commits this atomic operation. After `commit` is called, the resource has no more effect on the execution
+        /// and future `commit` calls will cause a fatal error.
+        /// The atomic operation is automatically committed in case `commit` was not called when the resource is dropped.
+        commit: func() -> ();
+    }
+
+    /// Gets the current retry policy associated with the worker
+    get-retry-policy: func() -> retry-policy;
+
+    /// Overrides the current retry policy associated with the worker. Following this call, `get-retry-policy` will return the
+    /// new retry policy.
+    set-retry-policy: func(new-retry-policy: retry-policy) -> ();
```

Comment:
In theory this could be a resource, too, where committing would restore the retry policy.

Reply:
Yes, I was thinking about it, but I think it is confusing if you create multiple ones simultaneously, as it is not guaranteed that you drop them in opposite order.